### PR TITLE
macOS: Added default sdk and deployment version environmental variables. (closes #1135)

### LINKED ISSUE #1135: cmake fails on macos 13
**Describe the bug**
When i'm trying to cmake happen this error

```
lj_arch.h:82:10: fatal error: 'TargetConditionals.h' file not found
#include "TargetConditionals.h"
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**To Reproduce**
cmake in bin directory

**Expected behavior**
cmake w/o errors

**Desktop (please complete the following information):**
 - OS: macos ventura 13.0.1
 - MBP M1, 16GB RAM
 - clang version: Apple clang version 14.0.0 (clang-1400.0.29.202)
 - OpenXRay build version 
 -  git commit: 96797aa25cbad2f296b2684c6ecf4641a690be65
 -  git branch: dev

### PR DESCRIPTION
**Pull request by johncurley**
cmake was failing for LuaJIT due to calling a compiler test that did not have the environmental variable set for the MacOS SDK. This pull request sets the variable values for the sysroot and a default deployment target as well as comments out an unused clean function.

### CODE REVIEW

File: Externals/LuaJIT-proj/luajit.cmake [hunk 1]
```diff
@@ -56,10 +60,10 @@ set(LUA_CPATH "LUA_CPATH" CACHE STRING "Environment variable to use as package.c
 set(LUA_INIT "LUA_INIT" CACHE STRING "Environment variable for initial script")
 
 # Clean unnecessary files in LuaJIT source directory
-execute_process(
-	COMMAND ${CMAKE_MAKE_PROGRAM} clean
-	WORKING_DIRECTORY ${LUAJIT_DIR}
-)
+#execute_process(
```
Comment (Masterkatze) [inline Externals/LuaJIT-proj/luajit.cmake]: Why is it commented out?

File: Externals/LuaJIT-proj/luajit.cmake [hunk 2]
```diff
@@ -189,7 +197,8 @@ if (WIN32)
 	set(LJVM_MODE peobj)
 elseif (APPLE)
 	if (CMAKE_OSX_DEPLOYMENT_TARGET STREQUAL "")
```
Comment (Masterkatze) [inline Externals/LuaJIT-proj/luajit.cmake]: `if ($ENV{MACOSX_DEPLOYMENT_TARGET} STREQUAL "")` ?

File: Externals/LuaJIT-proj/luajit.cmake [hunk 3]
```diff
@@ -189,7 +197,8 @@ if (WIN32)
...
-		message(FATAL_ERROR "Missing export MACOSX_DEPLOYMENT_TARGET=XX.YY")
+		#message(FATAL_ERROR "Missing export MACOSX_DEPLOYMENT_TARGET=XX.YY")
```
Comment (Masterkatze) [inline Externals/LuaJIT-proj/luajit.cmake]: `message(WARNING "Missing export MACOSX_DEPLOYMENT_TARGET=XX.YY, setting it to ${CMAKE_OSX_DEPLOYMENT_TARGET}")` ?

Comment (johncurley) [reply Externals/LuaJIT-proj/luajit.cmake]: > `message(WARNING "Missing export MACOSX_DEPLOYMENT_TARGET=XX.YY, setting it to ${CMAKE_OSX_DEPLOYMENT_TARGET}")` ?

I have removed the need for this check altogether as it is defined in the SDK by default which we have now pointed to. I'll need someone who has access to the git submodules to comment out 

ifeq (,$(MACOSX_DEPLOYMENT_TARGET))
    $(error missing: export MACOSX_DEPLOYMENT_TARGET=XX.YY)
endif

in the file Externals/LuaJIT/src/Makefile, line :320.

**Comment by johncurley** [comment]
Reverted checks for MACOS_DEPLOYMENT_TARGET to previous versions.

File: Externals/LuaJIT-proj/luajit.cmake [hunk 4]
```diff
@@ -191,7 +196,7 @@ elseif (APPLE)
 	if (CMAKE_OSX_DEPLOYMENT_TARGET STREQUAL "")
 		message(FATAL_ERROR "Missing export MACOSX_DEPLOYMENT_TARGET=XX.YY")
 	endif()
-
+
```
Comment (Xottab-DUTY) [inline Externals/LuaJIT-proj/luajit.cmake]:
```suggestion

```

File: Externals/LuaJIT-proj/luajit.cmake [hunk 5]
```diff
@@ -191,7 +196,6 @@ elseif (APPLE)
 	if (CMAKE_OSX_DEPLOYMENT_TARGET STREQUAL "")
 		message(FATAL_ERROR "Missing export MACOSX_DEPLOYMENT_TARGET=XX.YY")
 	endif()
```
Comment (Xottab-DUTY) [inline Externals/LuaJIT-proj/luajit.cmake]:
```suggestion
	endif()

```

**Pull request merged**